### PR TITLE
MM-1995 Moved the ability to rename a team into the team settings modal and fixed valet regression

### PR DESCRIPTION
**Pull request by rgarmsen2295**
This moves the functionality to rename a team into the team settings area within a new tab named "General Settings".  It also fixes a few issues involving the team settings modal, including the issues with the valet feature.  Finally it changes the valet feature switch from using the "on/off" buttons to using simple radio buttons to be consistent with the rest of the site.

### CODE REVIEW

File: web/react/components/team_general_tab.jsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        }
+
+        var data = {};
+        data.new_name = name;
+
+        client.updateTeamDisplayName(data,
+            function nameChangeSuccess() {
+                this.props.updateSection('');
+                $('#team_settings').modal('hide');
+                window.location.reload();
```
Comment (crspeller) [inline web/react/components/team_general_tab.jsx]: This is not a good user experience. Especially if I had, or wanted to change other settings. Don't think that this should be fixed in this PR. But this should be a ticket.

**Comment by crspeller** [comment]
Looks good. @rgarmsen2295 Needs rebase.

File: web/react/components/team_general_tab.jsx (hunk 1 above)
Comment (rgarmsen2295) [inline web/react/components/team_general_tab.jsx]: Yeah totally agree, was going to make the ticket after this was in.  I looked into trying to fix it with this but getting the sidebar header to update seemed problematic and I didn't want to try and change a bunch of code along with these changes.  Like you said, it's definitely a problem to be fixed later, just will take it's own time to do right.

**Comment by rgarmsen2295** [comment]
@crspeller rebase complete

**Comment by jwilander** [comment]
+1

**Pull request merged**